monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -824,7 +837,7 @@ object Task extends TaskInstances {
               atom.get match {
                 case null => ()
                 case current =>
-                  val update = current :+ value
+                  val update = value :: current
                   if (!atom.compareAndSet(current, update))
                     onSuccess(value)
                   else if (update.length == expected) {
```

Comment:
We cannot use CountDownLatch because it's not implementable for Scala.js - we could have a State(list, length) in that atomic, though at this point we need to measure the performance compared to a synchronized approach.